### PR TITLE
DDF-4293 Fixed race conditions with SimpleDateFormatter

### PR DESCRIPTION
**Pull request by bdeining**
#### What does this PR do?
This PR fixes some SimpleDateFormatter race conditions.
#### Who is reviewing it? 
@paouelle @Bdthomson 

#### Ask 2 committers to review/merge the PR and tag them here.

@coyotesqrl
@pklinef
#### How should this be tested?

Since the change is 1:1 with SimpleDateFormat, this should be covered by unit tests.
#### What are the relevant tickets?
[DDF-4293](https://codice.atlassian.net/browse/DDF-4293)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/spatial/kml/spatial-kml-transformer/src/main/java/org/codice/ddf/spatial/kml/transformer/DescriptionTemplateHelper.java [hunk 1]
```diff
@@ -50,7 +51,7 @@
               Metacard.THUMBNAIL,
               Metacard.CONTENT_TYPE_VERSION));
 
-  private DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
```
Comment (rzwiefel) [inline catalog/spatial/kml/spatial-kml-transformer/src/main/java/org/codice/ddf/spatial/kml/transformer/DescriptionTemplateHelper.java]: :question: Is there something else using it or can we get rid of the import?

File: catalog/spatial/kml/spatial-kml-transformer/src/main/java/org/codice/ddf/spatial/kml/transformer/DescriptionTemplateHelper.java [hunk 2]
```diff
@@ -50,7 +51,7 @@
...
+  private FastDateFormat dateFormat = FastDateFormat.getInstance("yyyy-MM-dd'T'HH:mm:ssZ");
```
Comment (paouelle) [inline catalog/spatial/kml/spatial-kml-transformer/src/main/java/org/codice/ddf/spatial/kml/transformer/DescriptionTemplateHelper.java]: ✏️ Can we mark it `final` to clarify it won't be changed later.

**Comment by bdeining** [comment]
build now

**Comment by coyotesqrl** [comment]
build now

**Pull request merged**